src/AsyncStorage.native.ts

```diff
@@ -0,0 +1,366 @@
+        });
+      }
+
+      const getRequest: MultiRequest = {
+        keys: keys,
+        callback: callback,
+        // do we need this?
+        keyIndex: _getKeys.length,
+        resolve: null as any,
+        reject: null as any,
+      };
+
+      const promiseResult = new Promise<readonly KeyValuePair[] | void>(
+        (resolve, reject) => {
+          getRequest.resolve = resolve;
+          getRequest.reject = reject;
+        }
+      );
+
+      _getRequests.push(getRequest);
+      // avoid fetching duplicates
+      keys.forEach((key) => {
+        if (_getKeys.indexOf(key) === -1) {
+          _getKeys.push(key);
+        }
+      });
+
+      return promiseResult;
+    },
+
+    /**
+     * Use this as a batch operation for storing multiple key-value pairs. When
+     * the operation completes you'll get a single callback with any errors.
+     *
+     * See http://reactnative.dev/docs/asyncstorage.html#multiset
+     */
+    multiSet: (keyValuePairs, callback) => {
+      checkValidArgs(keyValuePairs, callback);
+      return new Promise((resolve, reject) => {
+        keyValuePairs.forEach(([key, value]) => {
+          checkValidInput(key, value);
+        });
+
+        RCTAsyncStorage.multiSet(keyValuePairs, (errors?: ErrorLike[]) => {
+          const error = convertErrors(errors);
+          callback?.(error);
+          if (error) {
+            reject(error);
+          } else {
+            resolve();
+          }
+        });
+      });
+    },
+
+    /**
+     * Call this to batch the deletion of all keys in the `keys` array.
+     *
+     * See http://reactnative.dev/docs/asyncstorage.html#multiremove
+     */
+    multiRemove: (keys, callback) => {
+      return new Promise((resolve, reject) => {
+        keys.forEach((key) => checkValidInput(key));
+
+        RCTAsyncStorage.multiRemove(keys, (errors?: ErrorLike[]) => {
+          const error = convertErrors(errors);
+          callback?.(error);
+          if (error) {
+            reject(error);
+          } else {
+            resolve();
+          }
+        });
+      });
+    },
+
+    /**
+     * Batch operation to merge in existing and new values for a given set of
+     * keys. This assumes that the values are stringified JSON.
+     *
+     * **NOTE**: This is not supported by all native implementations.
+     *
+     * See http://reactnative.dev/docs/asyncstorage.html#multimerge
+     */
+    multiMerge: (keyValuePairs, callback) => {
+      return new Promise((resolve, reject) => {
+        RCTAsyncStorage.multiMerge(keyValuePairs, (errors?: ErrorLike[]) => {
+          const error = convertErrors(errors);
+          callback?.(error);
+          if (error) {
+            reject(error);
+          } else {
+            resolve();
+          }
+        });
+      });
+    },
+  };
+})();
+
+// Not all native implementations support merge.
+if (!RCTAsyncStorage.multiMerge) {
+  delete AsyncStorage.mergeItem;
+  delete AsyncStorage.multiMerge;
+}
```

Comment:
guess we could get rid of that, all platforms include this methods

Reply:
I'll leave it in for now. I want this change to be more or less a pure migration. We can remove it in a separate PR.